### PR TITLE
FIX Handle fields with square brackets

### PR DESCRIPTION
**Pull request by dhensby**
Potential fix to #5277 

A more robust way to handle array like field names

### CODE REVIEW

**Comment by tractorcow** [comment]
A little hard to follow... is there a test you can add that demonstrates where this fixes the issue?

**Comment by dhensby** [comment]
@tractorcow see https://github.com/silverstripe/silverstripe-framework/issues/5277

Do we even think we should support this kind of "broken" syntax for form fields?

**Comment by tractorcow** [comment]
Ok, looking over the code, I think that your solution is better than the existing one. The existing regexp code is kind of flakey, and has caused other problems in the past.

I suggest cleaning up with some unit tests and we can merge it in.

**Comment by dhensby** [comment]
I think this test provides suitable coverage: https://github.com/silverstripe/silverstripe-framework/blob/3.4.2/tests/forms/FormTest.php#L27-L65

As for "Should we support it" - I mean supporting this bad use of array syntax in requests and then reverse engineering it from the form names. Wouldn't dot notation be better?

**Comment by dhensby** [comment]
Test failures look related :P

**Comment by dhensby** [comment]
@tractorcow ready

**Comment by tractorcow** [comment]
Restarting behat tests...

**Comment by tractorcow** [comment]
Ok, I guess these behat issues are pre-existing.
